buildpipeline/perf-pipeline.groovy

```diff
@@ -0,0 +1,391 @@
+@Library('dotnet-ci') _
+
+// Incoming parameters.  Access with "params.<param name>".
+// Note that the parameters will be set as env variables so we cannot use names that conflict
+// with the engineering system parameter names.
+
+//--------------------- Windows Functions ----------------------------//
+
+def windowsBuild(String arch, String config, String pgo, boolean isBaseline) {
+    checkout scm
+
+    String pgoStash = ((pgo == "skiprestoreoptdata") ? "nopgo" : "pgo")
+    String baselineString = ""
+
+    // For baseline builds, checkout the merge's parent
+    if (isBaseline) {
+        baselineString = "-baseline"
+        bat "git checkout HEAD^^1"
+    }
+
+    bat "set __TestIntermediateDir=int&&.\\build.cmd ${config} ${arch} skipbuildpackages ${pgo}"
+
+    // Stash build artifacts. Stash tests in an additional stash to be used by Linux test runs
+    stash name: "nt-${arch}-${pgoStash}${baselineString}-build-artifacts", includes: 'bin/**'
+    stash name: "nt-${arch}-${pgoStash}${baselineString}-test-artifacts", includes: 'bin/tests/**'
+}
+
+// TODO: need to add the SAS token
+def windowsPerf(String arch, String config, String uploadString, String runType, String opt_level, String jit, String pgo, String scenario, boolean isBaseline) {
+    checkout scm
+    String baselineString = ""
+    if (isBaseline) {
+        baselineString = "-baseline"
+    }
+    dir ('.') {
+        unstash "nt-${arch}-${pgo}${baselineString}-build-artifacts"
+        unstash "benchview-artifacts"
+        unstash "metadata"
+    }
+
+    // We want to use the baseline metadata for baseline runs. We expect to find the submission metadata in
+    // submission-metadata.py
+    if (isBaseline) {
+        bat "move /y submission-metadata-baseline.json submission-metadata.json"
+    }
+
+    String testEnv = ""
+
+    if (jit == 'legacy_backend') {
+        testEnv = "-testEnv %WORKSPACE%\\tests\\legacyjit_x86_testenv.cmd"
+    }
+
+    String failedOutputLogFilename = "run-xunit-perf-scenario.log"
+
+    bat "py \".\\Microsoft.BenchView.JSONFormat\\tools\\machinedata.py\""
+    bat ".\\init-tools.cmd"
+    bat "run.cmd build -Project=\"tests\\build.proj\" -BuildOS=Windows_NT -BuildType=${config} -BuildArch=${arch} -BatchRestorePackages"
```

Comment:
I assume that this step is needed as we used to do it as part of the build, but now need to do the work explicitly as we do the build on a separate machine?

Reply:
Correct. It's because we build the tests on a different machine.